### ISSUE TITLE
Downloading youtube video doesn't work

### ISSUE THREAD

**Issue by AndrejGajdos** [axet/vget#31]
Hi, one year ago I used vget library to download youtube videos. If I try it know, it doesn't work. When I reach v.download(); method in example, it is working and doesn't end (like in loop). I use the last version of library 1.1.22. Could you confirm me that library works?

[so](http://stackoverflow.com/questions/27874413/download-youtube-videos-dont-work-in-vget)

**Comment by yayitswei** [comment]
The latest version is not working for me either.

@axet Thanks for making vget!

**Comment by davinkevin** [comment]
Hello, 

Same thing for me since I moved to the 1.1.22 3 days ago.

I've made a Gist with the test I use to check the download : https://gist.github.com/davinkevin/d89c166de465b813bc8d (you have the url I use inside the gist)

I reach the line https://github.com/axet/vget/blob/deccccdec4df2245ca2a1de9772827e12400d49c/src/main/java/com/github/axet/vget/vhs/YouTubeParser.java#L672 every times... not good for a rare case...

**Comment by davinkevin** [comment]
I've done much investigation (related to my previous issue) and it seems if the link is in http, the error occurs and if I use https links... all work fine

**Comment by davinkevin** [comment]
After some times (don't know why), the same issue occurs. 

I revert to the version 1.1.5 (the previous version I used) and everything seems to be good...

I'il try to find (if I have the time to) what is the highest working version for me.

Thanks

**Comment by reptilian-love** [comment]
The hanging normally happens when the client is unable to extract video information, in that case it tries again after 10 seconds, it gets stuck in this loop forever.... The downloads stop working yesterday in my application, I'm going to investigate this evening.

**Comment by reptilian-love** [comment]
VGET 1.1.22

    public void download(VGetParser user, final AtomicBoolean stop, final Runnable notify) throws Exception {
        if (targetFile == null && targetForce == null && targetDir == null) {
            throw new RuntimeException("Set download file or directory first");
        }

        try {
            if (empty()) { /* THIS ALWAYS RETURNS TRUE ALL THE TIME, BECAUSE THE VIDEO INFO IS ALWAYS NULL, this issue started  */
                extract(user, stop, notify); //
            }

**Comment by zeneck82** [comment]
Version 1.1.22 stopped working because of the line 484 in `com.github.axet.vget.vhs.YouTubeParser` class:
```
Pattern urlencod = Pattern.compile("\"url_encoded_fmt_stream_map\": \"([^\"]*)\"");
```
Regexp doesn't match, because of the space after the colon. Apparently YouTube changed the formatting of the JSON string - the space is not there anymore.

Regards

**Comment by axet** [comment]
Thanks @zeneck82 . Released 1.1.23
